src/applications/disability-benefits/686c-674/README.md

```diff
@@ -0,0 +1,71 @@
+
+## The back end code
+
+There are really two separate API endpoints that we use in the 686c form, one of them dealing with the check for a valid VA file number noted in the front end sections, and one that receives the form payload on submission. The endpoint for the VA file number check is located at `vets-api/app/controllers/v0/profile/valid_va_file_numbers_controller.rb` and the endpoint for the submission payload is at `vets-api/app/controllers/v0/dependents_applications_controller.rb`.
+
+### Valid VA file number check (back end)
+
+When the Veteran first lands on the Introduction page of the form a call is made to `vets-api/app/controllers/v0/profile/valid_va_file_numbers_controller.rb` where we use BGS as a data source to see if the current user has valid VA file number. We do this with the `valid_va_file_number_data` method call which simply takes the data returned by BGS for the current user and sees if the user has a `file_nbr` which is where BGS holds the file number. If the user has a valid VA file number we return `true`, otherwise we return `false`, to the front end.
+
+### Form submission
+
+
+After the Veteran fills out the form and submits the payload it is sent to `vets-api/app/controllers/v0/dependents_applications_controller.rb` on the back end. The main method we use in that controller is the `create` method, this method starts out by creating a `claim` object. The `claim` object is created with the `new` method of the `DependencyClaim`. We then check if for some reason the `claim` did not get created, meaning that the form payload has not passed our validation (more on this in a moment), we then log the error and raise an exception.
+
+After we create the claim we then process any attached documents, which are needed for some of the workflows in the form, and then using the dependents service we send the data to BGS. In addition to sending the data to BGS we also use `PDFtk` to create a PDF that we then send to VBMS. VBMS then uploads the PDF to the Veteran's eFolder (like a "my documents" folder at the VA for each Veteran). Once the claim has been submitted successfully, at this point we clear out the current saved claim data so that this claim doesn't stay in the Veteran's `save in progress` function on the form. We then render JSON to the front end.
+
+### After submission
+When we submit the claim using the `dependent_service` call inside `vets-api/app/controllers/v0/dependents_applications_controller.rb`, that service can kick off three separate jobs based on the payload of the form. Previously we talked about how the user can select different workflows through the form, some of them specific to the 21-686c and one of them specific to the 21-674 form. Based on which workflow the Veteran filled out we use the `dependent_service` to kick off a job specific to the 686c, and/ or a job specific to the 674, and a job specific to creating the PDF using `PDFtk` and sending it to VBMS (this third job will always be run).
```

Comment:
I like the mention of the separate workflows. Is there any sort of distinct workflow for the 527 integration?

 I noticed that form ID was mentioned at the top of the document, but nowhere else thereafter. It may be worth dedicating a few words to how / why the 527 can be derived from a 686c submission.

Reply:
There is no distinct workflow for the 527, it is just the pension questions and they will be submitted along with the 686c stuff. I think it might be helpful to point this out though.

Reply:
Agreed. Without proper context, the income questions on the form do feel a little random, especially because they aren't required. I think adding a note that answering those questions allows a 527 to be derived is worth it.